src/main/java/lc/kra/system/keyboard/GlobalKeyboardHook.java

```diff
@@ -132,19 +138,66 @@ public GlobalKeyboardHook(boolean raw) throws UnsatisfiedLinkError {
 	 * @param event A global key event
 	 */
 	private void keyPressed(GlobalKeyEvent event) {
+		
+		int virtualKeyCode = event.getVirtualKeyCode();
+		
+		// only add if key is not already hold down
+		if(!isKeyHoldDown(event))
+			holdDownKeyCodes.add(virtualKeyCode);
+		
 		for(GlobalKeyListener listener:listeners)
 			listener.keyPressed(event);
 	}
+	
 	/**
 	 * Invoke keyReleased (transition state TS_UP) for all registered listeners
 	 * 
 	 * @param event A global key event
 	 */
 	private void keyReleased(GlobalKeyEvent event) {
+
+		int virtualKeyCode = event.getVirtualKeyCode();
+
+		// redundant if - key should always be hold down if you can release it, but you never know
+		if(isKeyHoldDown(event))
```

Comment:
Same here, there is no need to check this before removing something from a Set. As you said it a) should never happen and b) even if so, the Set will not cause any problem if you try to remove something which is not in the set.

Reply:
done